### PR TITLE
HBASE-23081 Add an option to enable/disable rs group feature

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache9** [comment]
Not fully done yet. At least need to revert lots of UT changes when implementing HBASE-22971...

Let's see the pre commit result.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ HBASE-22514 Compile Tests _ |
| :green_heart: |  mvninstall  |   5m 57s |  HBASE-22514 passed  |
| :green_heart: |  compile  |   0m 56s |  HBASE-22514 passed  |
| :green_heart: |  checkstyle  |   1m 32s |  HBASE-22514 passed  |
| :green_heart: |  shadedjars  |   4m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 36s |  HBASE-22514 passed  |
| :blue_heart: |  spotbugs  |   4m 53s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   4m 50s |  HBASE-22514 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   5m 25s |  the patch passed  |
| :green_heart: |  compile  |   0m 55s |  the patch passed  |
| :green_heart: |  javac  |   0m 55s |  the patch passed  |
| :green_heart: |  checkstyle  |   1m 31s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m  1s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  17m  6s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
| :green_heart: |  findbugs  |   4m 47s |  the patch passed  |
||| _ Other Tests _ |
| :broken_heart: |  unit  | 154m 34s |  hbase-server in the patch failed.  |
| :green_heart: |  asflicense  |   0m 27s |  The patch does not generate ASF License warnings.  |
|  |   | 216m 29s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestMasterMetrics |
|   | hadoop.hbase.regionserver.TestRegionServerMetrics |
|   | hadoop.hbase.util.TestHBaseFsckReplication |
|   | hadoop.hbase.regionserver.TestRegionReplicasWithRestartScenarios |
|   | hadoop.hbase.master.TestRollingRestart |
|   | hadoop.hbase.TestNamespace |
|   | hadoop.hbase.regionserver.TestRegionOpen |
|   | hadoop.hbase.master.TestClusterRestartSplitWithoutZk |
|   | hadoop.hbase.master.TestMasterRestartAfterDisablingTable |
|   | hadoop.hbase.master.TestClusterRestart |
|   | hadoop.hbase.master.TestDLSFSHLog |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/691 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 20eaaa7980e1 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-691/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 15c28d0d98 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/1/testReport/ |
| Max. process+thread count | 4293 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 10 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| :green_heart: |  mvninstall  |   5m 31s |  HBASE-22514 passed  |
| :green_heart: |  compile  |   0m 56s |  HBASE-22514 passed  |
| :green_heart: |  checkstyle  |   1m 20s |  HBASE-22514 passed  |
| :green_heart: |  shadedjars  |   4m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 39s |  HBASE-22514 passed  |
| :blue_heart: |  spotbugs  |   4m 22s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   4m 20s |  HBASE-22514 passed  |
||| _ Patch Compile Tests _ |
| :broken_heart: |  mvninstall  |   2m 31s |  root in the patch failed.  |
| :broken_heart: |  compile  |   0m 38s |  hbase-server in the patch failed.  |
| :broken_heart: |  javac  |   0m 38s |  hbase-server in the patch failed.  |
| :green_heart: |  checkstyle  |   1m 18s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :broken_heart: |  shadedjars  |   3m 24s |  patch has 16 errors when building our shaded downstream artifacts.  |
| :broken_heart: |  hadoopcheck  |   1m 50s |  The patch causes 16 errors with Hadoop v2.8.5.  |
| :broken_heart: |  hadoopcheck  |   3m 43s |  The patch causes 16 errors with Hadoop v2.9.2.  |
| :broken_heart: |  hadoopcheck  |   5m 41s |  The patch causes 16 errors with Hadoop v3.1.2.  |
| :green_heart: |  javadoc  |   0m 33s |  the patch passed  |
| :broken_heart: |  findbugs  |   0m 39s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| :broken_heart: |  unit  |   0m 38s |  hbase-server in the patch failed.  |
| :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  34m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/691 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 80995f76ad61 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-691/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 15c28d0d98 |
| Default Java | 1.8.0_181 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-compile-hbase-server.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-compile-hbase-server.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-shadedjars.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-javac-2.9.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-javac-3.1.2.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-findbugs-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/testReport/ |
| Max. process+thread count | 93 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 10 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| :green_heart: |  mvninstall  |   5m 28s |  HBASE-22514 passed  |
| :green_heart: |  compile  |   0m 56s |  HBASE-22514 passed  |
| :green_heart: |  checkstyle  |   1m 23s |  HBASE-22514 passed  |
| :green_heart: |  shadedjars  |   4m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 39s |  HBASE-22514 passed  |
| :blue_heart: |  spotbugs  |   4m 22s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   4m 20s |  HBASE-22514 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   4m 51s |  the patch passed  |
| :green_heart: |  compile  |   0m 55s |  the patch passed  |
| :green_heart: |  javac  |   0m 55s |  the patch passed  |
| :green_heart: |  checkstyle  |   1m 20s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  15m 24s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| :green_heart: |  findbugs  |   4m 28s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  | 153m 17s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 210m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/691 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7d709805c02d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-691/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 15c28d0d98 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/3/testReport/ |
| Max. process+thread count | 4448 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-691/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
OK, good. All green.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/DisabledRSGroupInfoManager.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  @Override
+  public void removeServers(Set<Address> servers) throws IOException {
+    throw new DoNotRetryIOException("RSGroup is disabled");
+  }
+
+  @Override
+  public RSGroupInfo getRSGroupForTable(TableName tableName) throws IOException {
+    return null;
```
Comment (petersomogyi) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/DisabledRSGroupInfoManager.java]: Shouldn't it be RSGroupInfo.DEFAULT_GROUP? In getRSGroupOfServer we return DEFAULT_GROUP so it feels a bit inconsistent.

Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/DisabledRSGroupInfoManager.java]: This method has been marked as deprecated. Now rs group will be stored in table descriptor, here this method is only useful during migration, where we still store the rs group info for a table is RSGroupInfo. As here we disabled rs group feature, just return null here. The upper layer will use the rs group gotten from the table descriptor.

**Pull request merged**